packages/devtools/lib/src/timeline/frame_flame_chart.dart

```diff
@@ -202,4 +270,54 @@ class FrameFlameChart extends CoreElement {
     style.top = '${top}px';
     section.element.append(item);
   }
+
+  void _handleMouseMove(MouseEvent e) {
+    // Subtract offsets so that [currentMouseX] and [currentMouseY] reflect the
+    // (x, y) coordinates within the flame chart.
+    currentMouseX = e.client.x - element.offsetLeft;
+    currentMouseY = e.client.y - element.offsetTop;
+
+    // Store current scroll values for re-calculating scroll location on zoom.
+    currentScrollTop = element.scrollTop;
+    currentScrollLeft = element.scrollLeft;
+    currentScrollWidth = element.scrollWidth;
+  }
+
+  void _handleMouseWheel(WheelEvent e) {
+    e.preventDefault();
+
+    if (e.deltaY.abs() > e.deltaX.abs()) {
+      _handleZoom(e.deltaY);
+    } else if (e.deltaX.abs() > e.deltaY.abs()) {
+      // Manually perform horizontal scrolling.
+      element.scrollLeft += e.deltaX;
+      currentScrollLeft = element.scrollLeft;
+    }
+  }
+
+  void _handleZoom(num deltaY) {
+    if (deltaY > -0.0) {
+      _zoomIn();
```

Comment:
you should use the deltaY. With a larger deltaY the user is acting to zoom in or out faster.

Reply:
The deltaYs vary significantly between touchpad and mousewheel, which concerns me. The current state works well for both touchpad and mousewheel, so I'm going to leave it for now. We can always change it later if we decide the zooming is too fast/too slow.